Changes modid to ae2

shartte opened the pull request:
The diff here is somewhat misleading.
I made two commits. One moves the assets, the other makes the changes. This way Git doesn't detect the move as a delete/readd and keeps the per-line history for unchanged lines in the assets.
Githubs diff viewer doesn't agree though, so it'll seem as if all assets have changed, unless you limit the diff-view just to one commit.

yueh commented:
There is some serious question why not everything uses `AppEng.MOD_ID` or at least something like `AppEng.RESOURCE_FOLDER`. (But that is about the old code)

shartte commented:
I've changed it to the constant, where the ResourceLocation constructor was used. In other places, It actually becomes a problem with how misleading ModelResourceLocation's constructors are structured.
Personally? Resource locations are usually magic constants already, so refactoring out a portion of those (one that almost never changes) seems to bring little benefit.

Then again, we can change that in a different PR.

shartte commented:
Allright @elix-x, what's your take on this?

Elix-x commented:
Still kinda unsure about lenght we need.

Also, while we're here, we should consider removing major version from modid (`ae2` -> `ae`).

_Over on main thread, i ran a poll for versioning: [74% wants us to move to semantec versioning](http://www.strawpoll.me/11011534/r). Though changes to gradle should be done in different PR, this is a valid cause for removing major version from modid._

_@yueh when we were discussing it there, we left this question undecided because we didn't know what's better for users. We do now._

shartte commented:
@elix-x Hm, I could go for ae. I think what speaks for `ae2` is that actual players still use the abbreviation "AE2" often.

_edit_: What speaks for `ae`: If we decide to - in the far future - bring out AE3 with big changes, there still would be a possibility to provide an upgrade path for old worlds I guess.

Elix-x commented:
> edit: What speaks for ae: If we decide to - in the far future - bring out AE3 with big changes, there still would be a possibility to provide an upgrade path for old worlds I guess.

And that's exactly the reason.

shartte commented:
@yueh Your take on this? I think I can agree to `ae` and would amend the PR accordingly.

yueh commented:
I do not have any objections against removing the `2`. But no idea about just `ae`. Blocking a 2 char mod id is a bit presumptuous towards other mods.

I will ask around regarding a potential limit on the length for mod id (why the heck is forge not simply using `PreConditions`?). But in doubt `appliedenergistics` would probably also be fine. With lex I would not be surprised, if he decides one day to fuck with mods and enforce a 8 character minimum. Maybe just start with the resource location and 1 version later also enforced for the mod id.

Elix-x commented:
> With lex I would not be surprised, if he decides one day to fuck with mods and enforce a 8 character minimum

Well, as i said, pre 1.9, 3 character modids weren't possible due to a bug. Lex said it will never be fixed, but it was... No idea how and who.

shartte commented:
@elix-x @yueh Are both of you fine with just `appliedenergistics` (no 2 at the end)?
If so, I'll change it to that. Except for "ie2", and some FTB-internal mods, I haven't found any mod that use a short abbreviation of its name as the modid (i.e. "immersiveengineering", "bloodmagic", etc.). So I think you are right, we would probably get away with "ae", but it's pretentious.

Elix-x commented:
I am.

shartte committed:
Changed modid to "appliedenergistics".

shartte committed:
Moved assets from appliedenergistics2 -> appliedenergistics

shartte commented:
Allright, I changed the PR so it changes to `appliedenergistics` instead of `ae2` and rebased/squashed whatnot. It's still two commits so the Git diff (at least in the IDE) is clean.

shartte commented:
@elix-x Unless yueh has objections, this one should be mergable I hope.

Elix-x commented:
It is.

shartte commented:
Heh you're always so cryptic. Are you waiting on yueh's confirmation?

Elix-x commented:
Yes, but i do not live in a crypt :D <sub><sup><sub><sup>or do i 0_o ???</sup></sub></sup></sub>.

yueh commented:
I kinda hoped for a response regarding the length. But nothing for now.

After some pondering about `appliedenergistics` I am really questionable now. The likelihood of AE2 ever being compatible with AE3 is next to nothing. AE3 would only be valid with major game changes, which certainly would be incompatible with any previously existing things. going the enderio way and feature bloating AE2 would still be AE2. AE3 would have to be something new. Something silly like moving everything to their own spatial dimenson, which is a copy of the normal dimension and you can place cables into the same blockspace. No overworld facades or even cables anymore.

AE2 is really the name of AE2 and not just a version. It is an actual successor to AE, thus really a different mod. If we apply semver to AE2, we would actually be somewhere at AE2 4.x.x now and after the 1.10 port probably AE2 5.0.0

Elix-x commented:
Yes, but will we want to write a separate mod to port AE2 worlds, like you algo had with AE1 worlds?

shartte commented:
It is fine with me, but we should make a decision quickly since this is holding up more work.
At this point, I am kinda indifferent about the modid, so I wouldn't be opposed to completely scrapping this PR.

yueh commented:
The separate mod was mostly to convert the old storage format to the new one. The problem this pretty much only works with vanilla items and blocks as these are the only constant ones. Currently you cannot even copy anything from world A to world B with the exact same verion of every mod.

I am not even aware that anyone actually used it.

---

Edit

I would consider putting it on hold is the best option for now. It is really not that important and it does not block any other change. As long as there is not beta release, I do not even have a problem with changing it midway and breaking existing dev worlds. If we just refactor it to use global constants instead of string copied everywhere it is pretty easy to change. The recipes further already support aliases, thus they are already using `ae2` as mod id instead of `appliedenergistics2`. Ideally it would just be renaming the folder and change 2 lines of code.

shartte commented:
Closing this for now.
I think we will stay at `appliedenergistics2` until something like AE3 comes along.
